### DISCUSSION TITLE
Compiler Error

### DISCUSSION THREAD

**Discussion by kaleb-keny** [vyperlang/vyper#2248] [Q&A] [answered]
Thank you @fubuloubu for setting this up. Hopefully this is a good way to keep track of questions, so that other people who might have similar questions can find answers instead of looking inside closed issues.
So I am facing an issue with the vyper compiler and `[Compiler Error] Error executing vyper: 'vyper' is not recognized as an internal or external command, operable program or batch file`. So I never was able to have the auto-compiler work on VS Code, therefore I resort to brownie for compilation. Is this normal, or should the vy extension be able to autocompile the code as I write it?
![image](https://user-images.githubusercontent.com/62237347/101683059-b99ac280-3a64-11eb-8c7f-6b76c811c1bf.png)

**Comment by fubuloubu** [comment]
Seems like a good task for https://github.com/vyperlang/vscode-vyper

**Reply by kaleb-keny** [reply]
Yeah I mean I installed the extension on the market place... https://marketplace.visualstudio.com/items?itemName=tintinweb.vscode-vyper through the vs code store...  But the extension might not be loading right... No worries I'll have a look at the gh repository to see if it's a requirement I'm missing. Thank you

**Reply by fubuloubu** [reply]
We forked it from this upstream repository, so would be a good place to start: https://github.com/tintinweb/vscode-vyper

**Comment by mat774** [comment, marked as answer]
I had the same issue today on my new PC. The cause and solution were in my case the following (I'm working on a Windows PC):

Cause: 
- I used a virtual environment for my contract delevopment, therefore Vyper was only installed in the venv (by using pip install vyper).
- Therefore, the vyper extension in VS Code was trying to use the shell command vyper to compile.
- But this command only exists in the venv.

Solution:
- Go to your venv folder and look for the vyper.exe file (venv -> scripts -> vyper.exe)
- Copy the path to the exe file
- Go in VS Code -> Settings -> Extensions -> Vyper -> Change vyper Command to the path (e.g. C:\dev\your_project\.venv\Scripts\vyper.exe)
- restart VS Code

Worked for me. Hope for you too.

**Reply by kaleb-keny** [reply]
I will try it out, thank you! also will link this in the upstream repo 🙏

**Reply by kaleb-keny** [reply]
ohh you did already thanks a lot

**Reply by kaleb-keny** [reply]
to update it worked... as @mat774  described exactly

**Reply by slayde704** [reply]
I'm glad this worked our for you. Please select my anwer, so it will displayed on the top.

**Reply by ericdwkim** [reply]
Thank you, this fixed my issue :)

**Reply by Dekubaka** [reply]
Even years later you have helped this poor soul. Thank you!

**Reply by liam-ot** [reply]
works like a charm, on MacOS path is likely `/Users/you/venv/bin/vyper`

**Comment by seekingtau** [comment]
Just tacking on to this - I've had the same issue with VS code on Mac. I didn't install in a virtual environment, but VS Code is still returning the "[Compiler Error] Error executing vyper: /bin/sh: vyper: command not found". I've tried uninstalling and reinstalling the plugin, reinstalling vyper, to no luck. Any ideas?

**Reply by seekingtau** [reply]
Fixed the error by tracing the compiler and changing the compile command (basically what is done above). Thank you!

**Comment by 0x0077** [comment]
It works for me!!!!!!!!macbook

**Comment by josephabickel** [comment]
Worked in 2025!
